ql/src/java/org/apache/hadoop/hive/ql/exec/tez/ReduceRecordSource.java

```diff
@@ -152,10 +152,11 @@ void init(JobConf jconf, Operator<?> reducer, boolean vectorized, TableDesc keyT
     this.tag = tag;
 
     try {
-      inputKeyDeserializer = ReflectionUtils.newInstance(keyTableDesc
-          .getDeserializerClass(), null);
-      SerDeUtils.initializeSerDe(inputKeyDeserializer, null, keyTableDesc.getProperties(), null);
-      keyObjectInspector = inputKeyDeserializer.getObjectInspector();
+      AbstractSerDe serde = ReflectionUtils.newInstance(keyTableDesc.getSerDeClass(), null);
+      serde.initialize(null, keyTableDesc.getProperties(), null);
+
+      inputKeyDeserializer = serde; 
```

Comment:
I suggest to call it inputKeySerDe, and it can be created under this name, no need to call it serde for an interim time.

Reply:
Done.  I had it this way because `inputKeyDeserializer` is a `Deserializer` class which does not have an 'initialize' method.  However, I change the type to be `AbstractSerDe` to address your comments here.